tests/unittests/clients/studio/resources/assistant/plans/test_plan_body_creation.py

```diff
@@ -0,0 +1,130 @@
+from typing import Callable, List, Dict, Any, Type, Union
+
+from pydantic import BaseModel
+from ai21.clients.common.beta.assistant.plans import BasePlans
+from ai21.errors import CodeParsingError
+from ai21.models.responses.plan_response import PlanResponse, ListPlanResponse
+from ai21.types import NotGiven, NOT_GIVEN
+import pytest
+
+
+class PlanTestClass(BasePlans):
+    def create(
+        self,
+        *,
+        assistant_id: str,
+        code: Union[str, Callable],
+        schemas: Union[List[Dict[str, Any]], List[Type[BaseModel]], NotGiven] = NOT_GIVEN,
+        **kwargs,
+    ) -> PlanResponse:
+        pass
+
+    def list(self, *, assistant_id: str) -> ListPlanResponse:
+        pass
+
+    def retrieve(self, *, assistant_id: str, plan_id: str) -> PlanResponse:
+        pass
+
+    def modify(
+        self, *, assistant_id: str, plan_id: str, code: str, schemas: Union[List[Dict[str, Any]], NotGiven] = NOT_GIVEN
+    ) -> PlanResponse:
+        pass
+
+
+def test_create_body__when_pass_code_str__should_return_dict():
+    # Arrange
+    code = "code"
+
+    # Act
+    result = PlanTestClass()._create_body(code=code)
+
+    # Assert
+    assert result == {"code": code}
+
+
+def test_create_body__when_pass_code_callable__should_return_dict():
+    # Arrange
+    def code():
+        return "code"
+
+    # Act
+    result = PlanTestClass()._create_body(code=code)
+
+    # Assert
+    assert result == {"code": 'def code():\n        return "code"'}
+
+
+def test_create_body__when_pass_code_and_dict_schemas__should_return_dict_with_schemas():
+    # Arrange
+    code = "code"
+    schemas = [{"type": "object", "properties": {"name": {"type": "string"}}}]
+
+    # Act
+    result = PlanTestClass()._create_body(code=code, schemas=schemas)
```

Comment:
Tests look great. Why not call the public `create` method? its better practice to test the public function, allowing us to change the logic to the private ones without breaking the tests.